### PR TITLE
GEODE-9285: Implement accurate sizeable for RedisSet

### PR DESCRIPTION
**Pull request by DonalEvans**
- Extend ObjectOpenCustomHashSet to allow direct access to the current
 capacity of the backing array
 - Calculate the overhead associated with each member added based on its
 length
 - Calculate the overhead associated with resizing the backing set in
 RedisSet
 - Update tests to reflect new behaviour

Authored-by: Donal Evans <doevans@vmware.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [N/A] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by nonbinaryprogrammer — CHANGES_REQUESTED** [review]
looks solid, I just have a couple of small suggestions.
did you run any internal tests for this?

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/SizeableObjectOpenCustomHashSet.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      return 16 + ((double[]) o).length * 8;
+    }
+    // If we get here, we can't figure out the size without using more expensive operations, so just
+    // give up
+    return 0;
+  }
+
+  private static int roundToMultipleOfEight(int arraySize) {
+    return ((arraySize + 7) / 8) * 8;
+  }
```
[the remark this answers is not in the document]
Comment (DonalEvans) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/SizeableObjectOpenCustomHashSet.java:168]: Nice catch, thanks!

**Review by nonbinaryprogrammer — CHANGES_REQUESTED** [review]
looks good but I think you missed this comment: https://github.com/apache/geode/pull/6531/files/2af372b1bc7a9fd58e5361b8e2289de659461753#r641816516
I'll approve it with that addition

**Comment by DonalEvans** [comment]
> looks good but I think you missed this comment: https://github.com/apache/geode/pull/6531/files/2af372b1bc7a9fd58e5361b8e2289de659461753#r641816516
> I'll approve it with that addition

Ah, my bad. I originally determined the equation experimentally, not exactly via a test but rather by inspection of the numbers and some educated guesswork, but I've since added a test, `backingArrayOverheadCalculationTest()` that should help determine the formula if it changes.

**Review by sabbey37 — APPROVED** [review]
Looks great!

**Review by nonbinaryprogrammer — APPROVED** [review]

**Pull request merged**